File: backend/ChatApp.Domain/Entities/Call.cs
using ChatApp.Domain.Common;
using ChatApp.Domain.Enums;

namespace ChatApp.Domain.Entities;

public class Call : BaseEntity
{
    public Guid ConversationId { get; set; }

    public Conversation Conversation { get; set; } = null!;

    public Guid CallerId { get; set; }

    public User Caller { get; set; } = null!;

    public Guid? ReceiverId { get; set; }

    public User? Receiver { get; set; }

    public CallStatus Status { get; set; } = CallStatus.Ringing;

    public DateTime? AcceptedAt { get; set; }

    public DateTime? EndedAt { get; set; }

    public int? DurationInSeconds { get; set; }
}

File: frontend/chat-app-client/src/types/call.ts
export enum CallStatus {
  Ringing = 1,
  Accepted = 2,
  Rejected = 3,
  Missed = 4,
  Ended = 5,
  Failed = 6,
}

export interface StartCallRequest {
  conversationId: string;
  receiverId: string;
}

export interface CallResponse {
  id: string;
  conversationId: string;
  callerId: string;
  callerUsername: string;
  receiverId?: string | null;
  receiverUsername?: string | null;
  status: CallStatus;
  createdAt: string;
  acceptedAt?: string | null;
  endedAt?: string | null;
  durationInSeconds?: number | null;
}

export interface WebRtcSignalRequest {
  callId: string;
  targetUserId: string;
  data: string;
}